src/Symfony/Component/JsonPath/JsonPathUtils.php

```diff
@@ -85,4 +85,80 @@ public static function findSmallestDeserializableStringAndPath(array $tokens, mi
             'tokens' => $remainingTokens,
         ];
     }
+
+    public static function unescapeString(string $str, string $quoteChar): string
+    {
+        if ('"' === $quoteChar) {
+            // try JSON decoding first for unicode sequences
+            $jsonStr = '"' . $str . '"';
+            $decoded = json_decode($jsonStr, true);
+
+            if (null !== $decoded) {
+                return $decoded;
+            }
+        }
+
+        $result = '';
+        $length = strlen($str);
```

Comment:
can you run php-cs-fixer on your side please, since fabbot is on holidays?
```suggestion
        $length = \strlen($str);
```